eventauth.go

```diff
@@ -801,13 +795,17 @@ func (a *allowerContext) redactEventAllowed(event PDU) error {
 	// sender and the redacted event.
 	// We leave it up to the sending server to implement the additional checks
 	// to ensure that only events that should be redacted are redacted.
-	if senderDomain == redactDomain {
+	sender, err := event.UserID()
+	if err != nil {
+		return err
+	}
+	if string(sender.Domain()) == redactDomain {
```

Comment:
This check doesn't make sense in room versions 3+ because event IDs are hashes, so `redactDomain` will be nonsensical, which is why we return early on `:783`. As we are basing pseudo IDs on modern room versions, the fact this uses domains and user IDs is fine.